### PR TITLE
Adding two R notebook to examples repo

### PR DESCRIPTION
**Pull request by lidefi87**
The first notebook [Extract_Water_Temp](https://github.com/lidefi87/rimrep-examples/blob/main/poc-data-api/Extracting_Water_Temperature_at_Site.md) uses the AIMS SST data and extracts data for points of interest. The second notebook, [Extract_Spatial_Data_GBR](https://github.com/lidefi87/rimrep-examples/blob/main/poc-data-api/Extracting_Spatial_Data_GBR_Features.md), extracts the spatial data for all GBR features in the Great Barrier Reef (GBR) Feature and creates a shapefile that can be saved locally.

There are two versions of each example script: the GitHub compatible version (ending in .md) and the RStudio friendly version (.Rmd). I have kept both because they may be useful to different users, but I should probably add some information about what each format is better for in the `README` file. But we can keep a single version if you think it is best.

### CODE REVIEW

**Review by barbuz — COMMENTED** [review]
The notebooks look great, nice job!

Would it be possible to have the generated maps show in the markdown notebooks? It would be nice as a way to show everything without requiring people to run the code themselves.

I have just a couple of notes about our datasets:
- You have used `UNIQUE_ID` for the gbr-features, but that's not actually the ID column of the dataset. I am not sure myself how exactly that dataset has been built, but it has multiple columns with "ID" in the name which all have some duplicate values and so cannot act as IDs for the rows. When no valid ID column is present in the dataset we will always add a new column `fid` with generated unique ID values, as we did in this case. I don't think you really use the value of the ID column here so it doesn't matter, I just wanted to let you know.
- The aims-sst dataset has two columns `lat` and `lon` that you could use instead of extracting this information from the geometry column. It's still good that you did that, to show people how they can manipulate geometries. In general, we will always have 4 columns with the minimum and maximum lat/lon (as we need them for pygeoapi), but for point geometries such as here, 2 columns are enough.

**Comment by lidefi87** [comment]
Thanks! I just added the figures to the notebooks. These will only show in the `.md` files though. Should I add a note in the `README` file explaining why we have two versions of these files?

About the IDs for the GBR features, I was a little confused myself, but I used the `UNIQUE_ID` field because it was the only one that matched the description of the unique code used by the GBRMPA (10-11 digits code). I kept the names because I thought it would be useful for some users. But, I will definitely create a new ID column if I ever need it.

About the AIMS SST data, should I use the `lat` and `lon` fields instead? I think I may have misunderstood you, I thought you wanted to use the `geometry` field. Looking at the datasets again, I think it was only for the GBR data. I can add the `lat` and `lon` option to this notebook, so people can see that they have two options to access the spatial data.

About the min and max coordinates, I can add this to the GBR example. I saw these fields, but did not access them as the `geometry` field had more detail.

**Comment by barbuz** [comment]
> Should I add a note in the README file explaining why we have two versions of these files?

Yep, that would be nice!

> But, I will definitely create a new ID column if I ever need it.

You don't need to create a new ID column, I just meant that we've already done it and it's called `fid` in the dataset! 
I agree, getting the codes used by GBRMPA is going to be useful for some users, so I'm happy with the way the notebook does it now. 

> About the AIMS SST data, should I use the lat and lon fields instead? I think I may have misunderstood you, I thought you wanted to use the geometry field.

No, you are right, we want these notebooks to show how you can use the geometries included in the data. But the lat/lon (or min/max) columns can be much easier to use for some purposes. Where the geometries columns are really useful is when you want to plot the areas covered by the data (as you have done in the gbr-features notebook), or when you want to select data that is part of a certain region. 
We could think of an example where you select a specific reef from gbr-features, get its geometry, and use the geometry to filter the aims-sst data to keep only the points inside this reef. But we don't need to have all that in these notebooks, it can be the subject of a future one.

In short, maybe you could add a mention about how the same information in this case could be read from the lat/lon columns, but I think these notebooks are generally good to go.

**Comment by diodon** [comment]
Great job Denisse!!! Very well documented. This is what we need, a narrative that explains the process in detail. 

Regarding the extraction of the data by site, this project has sites as the reef name where the loggers are deployed and sub-sites (also known as locations) which are the location on the reef. They normally deploy in the reef flat (<3m) and on the reef slope. If you explore the subsite name, you will find SL and FL. Maybe is a good idea to separate the calculation of the mean temperature by location

**Comment by diodon** [comment]
Could you extend the GRB features notebook to use one of the geometries for one Reef and filter the AIMS temperature dataset to extract data inside the reef's polygon? Maybe is better to make another notebook

Maybe in the form of a function that receives the reef name or the reef id and returns the polygon as an sf object that could be used as a boundary for filtering

**Comment by lidefi87** [comment]
I updated the `README` with more information about the two markdown formats with the same name. 

> You don't need to create a new ID column, I just meant that we've already done it and it's called `fid` in the dataset! I agree, getting the codes used by GBRMPA is going to be useful for some users, so I'm happy with the way the notebook does it now.

Ah, I got it now. I did not notice there was a `fid` field in the GBR dataset too.

> In short, maybe you could add a mention about how the same information in this case could be read from the lat/lon columns, but I think these notebooks are generally good to go.

I have added this information to the AIMS notebook. I labelled them option 1 and 2 for site and coordinate extraction. Hopefully it is clear that this shows two different ways of obtaining the same information. If this is not clear, let me know and I will improve the text.

I will be working on @diodon comments now. I will keep you updated.

**Comment by lidefi87** [comment]
> Regarding the extraction of the data by site, this project has sites as the reef name where the loggers are deployed and sub-sites (also known as locations) which are the location on the reef. They normally deploy in the reef flat (<3m) and on the reef slope. If you explore the subsite name, you will find SL and FL. Maybe is a good idea to separate the calculation of the mean temperature by location

I am having trouble finding the field that contains this information. I looked through all fields and I could not find any classification containing SL/slope and FL/flat or depth. But there are two fields that may be relevant: `CLASS_CONF`, which has four classes ranging from low to high, and `UN_FEATURE`, which is a boolean field.

Yesterday, I tried to find this dataset in stac and pygeoapi, but I could not find it, but I think I found the [source](https://eatlas.org.au/data/uuid/ac8e8e4f-fc0e-4a01-9c3d-f27e4a8fac3c). The fields described there do not have information about the deployment depth, but the two fields I highlighted above are not included in this website. It'd be great if I could get some finding the right field I should be using for this.

**Comment by diodon** [comment]
> > Regarding the extraction of the data by site, this project has sites as the reef name where the loggers are deployed and sub-sites (also known as locations) which are the location on the reef. They normally deploy in the reef flat (<3m) and on the reef slope. If you explore the subsite name, you will find SL and FL. Maybe is a good idea to separate the calculation of the mean temperature by location
> 
> I am having trouble finding the field that contains this information. I looked through all fields and I could not find any classification containing SL/slope and FL/flat or depth. But there are two fields that may be relevant: `CLASS_CONF`, which has four classes ranging from low to high, and `UN_FEATURE`, which is a boolean field.
> 
> Yesterday, I tried to find this dataset in stac and pygeoapi, but I could not find it, but I think I found the [source](https://eatlas.org.au/data/uuid/ac8e8e4f-fc0e-4a01-9c3d-f27e4a8fac3c). The fields described there do not have information about the deployment depth, but the two fields I highlighted above are not included in this website. It'd be great if I could get some finding the right field I should be using for this.

Sorry, there is no variable identifying the location. "FL" and "SL" (and some others too) are part of the `subsite` string: 

```
location$subsite[grepl("SL", location$subsite)]
  [1] "KIMSL1"            "LOWENDALSL1"       "LOWENDALSL1S"      "BRIDLEDSL1"        "VANSL1"            "LIMDSL1"           "LIMSL1"            "LIMVDSL1"          "TUARSL1"          
 [10] "RAISL1"            "SANSL1"            "SANDSL1"           "TYDSL1"            "NONAMESL1"         "MACGSL1"           "LIZSL1"            "COCSL1"            "BLASL1"           
 [19] "SNAPISSL1"         "LOWSL1"            "OPALSL1"           "AG1SL1"            "FRNKSL2"           "MCCULLSL1"         "SCOTT_SL4"         "CARSL1"            "IMPERIEUSEOUTSL1" 
 [28] "HIGHSL2"           "FITZSL2"           "THETSL1"           "ARLSL1"            "GRESL1"            "NBARSL1"           "DUNKSL1"           "TAYLORSL1"         "MYRSL1" 
```

So you need to parse the string and create a new variable (e.g. `location`)

**Comment by lidefi87** [comment]
> Regarding the extraction of the data by site, this project has sites as the reef name where the loggers are deployed and sub-sites (also known as locations) which are the location on the reef. They normally deploy in the reef flat (<3m) and on the reef slope. If you explore the subsite name, you will find SL and FL. Maybe is a good idea to separate the calculation of the mean temperature by location

I realised that I was looking at the wrong data, the GBR features instead of the AIMS temperature. This is now completed. But, I should point out that there were sites that did not have either of the deployment tags: `SL` or `FL`. I classified these as `NA`, but we can create a new category, perhaps `unknown` will be better. If you have a preference on how to classify this group with no deployment information, let me know and I will update this notebook.

**Comment by lidefi87** [comment]
> Could you extend the GRB features notebook to use one of the geometries for one Reef and filter the AIMS temperature dataset to extract data inside the reef's polygon? Maybe is better to make another notebook
> 
> Maybe in the form of a function that receives the reef name or the reef id and returns the polygon as an sf object that could be used as a boundary for filtering

This has now been completed. I created a few functions that allow users to select GBR features by name, ID or both. There is also a new notebook showing how this works.

I will not be off until next week. I will continue with the Python notebooks then.

File: poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+library(stringr)
+library(ggplot2)
+library(sf)
+library(lubridate)
+source("useful_spatial_functions_data_extraction.R")
+```
+
+# Connecting to RIMReP collection
+
+Connecting to the AIMS Sea Surface Temperature Monitoring Program. This can take a minute or so.
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd:34]: Connecting to the AIMS Sea Surface water temperature monitoring Program dataset

File: poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+- `LOC_NAME_S`, combination between feature name and unique ID
+- `geometry`, which includes latitude and longitude coordinates in [well-known binary (WKB)](https://loc.gov/preservation/digital/formats/fdd/fdd000549.shtml) format
+  
+We will transform the information in the `geometry` field into coordinate pairs (i.e., latitude and longitude in degrees for each node along its boundary).  
+  
+**Note:** One location (`GBR_NAME`) may contain multiple values under the `UNIQUE_ID` column. This is above water structures may not appear connected when looking at the surface, but they are part of the same structure and thus are connected under water.
+  
+## Extracting sites and coordinates from dataset
+
+We can extract data from the AIMS dataset by using `dplyr` verbs as shown below.
```
Comment (diodon) [inline poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd]: AIMS Sea Surface Water Temperature dataset by using `dplyr` verbs as shown below. This could take few seconds up to one minute depending on your Internet connection speed.

File: poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+sites <- data_df %>% 
+  #We select unique sites included in the dataset
+  distinct(UNIQUE_ID, GBR_NAME, LOC_NAME_S, geometry) %>%
+  #This will load them into memory
+  collect()
+
+#Checking results
+glimpse(sites)
+```
+As explained above, the `geometry` field is in WKB format, which we will transform into degree coordinates in the next step. Here, we will use the `WKB` to transform the `geometry` and then we will convert it to an `sf` object for easy data manipulation.
```
Comment (diodon) [inline poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd]: Here, we will use the `readWKB` function from the `WKB` package to transform the `geometry`

File: poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+  rename("geometry" = "coords_deg") %>% 
+  #Transforming into shapefile
+  st_as_sf() %>% 
+  #Assigning reference systems: WGS84 (EPSG: 4326)
+  st_set_crs(4326)
+
+#Checking results - We will exclude the geometry column
+head(sites)
+```
+
```
Comment (diodon) [inline poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd]: Add here more info about the process of transforming the WKB geometry into an SF polygon object and the advantages of this object for plotting and filtering

File: poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+## Plotting all features 
+
+We will make a map of all features included within the Great Barrier Reef Marine Protected Area in green so they are easily recognised. Note that the GBR features also includes a `Mainland` location, which represents the state of Queensland. We will show this state as light red.
+
+```{r map_sites}
+#Loading basemap with countries in Oceania and Asia
+australia <- rnaturalearth::ne_countries(country = "Australia", returnclass = "sf")
```
Comment (diodon) [inline poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd]: here you assume that `rnaturalearth` package is available. Include it in the import packages section

File: poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+```{r libraries, results = "hide", message = F, warning = F}
+library(arrow)
+library(dplyr)
+library(magrittr)
+library(stringr)
+library(wkb)
+library(ggplot2)
+library(sf)
+library(lubridate)
+```
```
Comment (diodon) [inline poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd]: `library(rnaturalearth)`

File: poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+# Goal of this notebook
+
+This notebook will show how to access the RIMReP `geoparquet` collection for Great Barrier Reef (GBR) Feature from the Great Barrier Reef Marine Park Authority (GBRMPA). This dataset includes the unique IDs and names of all features above water, including sand banks, reefs, cays, islets, and islands. Since this dataset includes spatial data, we can extract the spatial limits of each feature included in this dataset.
+
```
Comment (diodon) [inline poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd]: Maybe adding some steps: 

1. Connect with RIMReP_DMS GBR features data collection and explore it
2. Get GBR features geometries
3. Convert WKB geometries into Simple Feature (sf) object
4. Export polygons as shapefiles

File: poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+# Goal of this notebook
+
+This notebook will demonstrate how to use one or more Great Barrier Reef (GBR) features recognised by the Great Barrier Reef Marine Park Authority (GBRMPA) dataset to extract water temperature data from the AIMS Sea Surface Temperature Monitoring Program available at RIMReP `geoparquet` collection.
+
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd]: This notebook uses a set of functions specially created for manipulating RIMReP data collections. The file `useful_spatial_functions_data_extraction.R` must exist in the same directory of this notebook

File: poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd (hunk 1 above)
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd]: AIMS Sea Surface Temperature Monitoring Program dataset

File: poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+  #This will load them into memory
+  collect()
+
+#Creating shapefile of unique locations
+sites_shp <- sites_shp %>% 
+  #Creating column to identify deployment location
+  mutate(deployment_location =  case_when(str_detect(subsite, "FL[0-9]{1}") ~ "reef flat",
+                                          str_detect(subsite, "SL[0-9]{1}") ~ "reef slope",
+                                          #If no condition is met, return NA
+                                          T ~ NA)) %>%
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd]: return "Other" instead of NA

File: poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+  #Turning into shapefile
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd]: turning into a `sf` object

File: poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+  st_as_sf(coords = c("lon", "lat"), crs = 4326)
+
+#Checking results
+glimpse(sites_shp)
+```
+# Identifying AIMS sites within GBR feature of interest
+In this section, we will use the functions included in the [useful_spatial_functions_data_extraction](https://github.com/aodn/rimrep-examples/blob/main/poc-data-api/useful_spatial_functions_data_extraction.R) script to load GBR features of interest. We will then use the boundaries of the feature of interest to identify the names of the sites within these limits.
+
+We will use the `gbr_features` function to access the boundaries for our sites of interest. If no parameters are given to this function, it will load all features recognised by the GBRMPA. You could also provide the name of the site using the `site_name` parameter or the GBRMPA unique ID using the `site_id` parameter. You can also use both parameters at the same time in case you have the names for some sites and the IDs for others.
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd]: it will load all features recognised by ~the~ GBRMPA.

File: poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+
+As an example, we will use **U/N Reef** or unnamed reefs. We chose them simply because they cover a large area and will overlap with the most number of AIMS monitoring sites.
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd]: In fact there are multiple reefs without a name, and they are referred to as **U/N Reef**, but each of them has a `UNIQUE_ID` value. The use of **U/N Reef** as a filter will return multiple reefs in the GBR

File: poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+Depending on the amount of polygons used to identify the sites of interest, this may take a couple of minutes.  
+  
+We will plot the original AIMS monitoring sites (black), the polygon of interest (pink) and the sites within polygons (blue) to check that sites have been correctly identified.
+
+Note that unless the AIMS site is within a polygon, it will not to be identified in our final list.
+
+```{r sites_int}
+site_list <- sites_of_interest(sites_shp, un_reefs)
+
+#Plotting results
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd]: How about plotting using Leafleat instead, so the user can zoom in/out the map

File: poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+  #Polygon of interest
+  geom_sf(inherit.aes = F, data = un_reefs, color = "pink")+
+  #AIMS polygons within area of interest
+  geom_sf(inherit.aes = F, data = site_list, color = "blue")+
+  theme_bw()
+```
+
+## Extracting temperature for monitoring site within area of interest
+
+We will now extract data only for the monitoring sites located within the GBR feature of our interest.
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd]: This next step takes a lot of time to execute. Would it be better if you extract the sites of interest based on the previous step and then do the calculation locally?

File: poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd [hunk 15]
```diff
@@ -0,0 +1,166 @@
...
+  dir.create(folder_out)
+}
+
+#Saving data summaries
+data_out <- paste(folder_out, "monthly_means_temperature.csv", sep = "/")
+write_csv_arrow(temp_area_int, file = data_out)
+
+#Saving plot
+plot_out <- paste(folder_out, "monthly_means_timeseries.tif", sep = "/")
+ggsave(plot_out, temp_plot, device = "tiff")
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd]: maybe saving as png instead of tiff, it will take less space

File: poc-data-api/Extracting_Water_Temperature_at_Site.Rmd [hunk 16]
```diff
@@ -0,0 +1,251 @@
...
+    html_preview: false
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+# Goal of this notebook
+
+This notebook will demonstrate how to access the RIMReP `geoparquet` collection for AIMS Sea Surface Temperature Monitoring Program. This dataset includes data since 1980 for tropical and subtropical coral reefs around Australia, including 80 Great Barrier Reef sites.
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_at_Site.Rmd]: I think there are more than 80 sites, around 300. Check.

File: poc-data-api/Extracting_Water_Temperature_at_Site.Rmd [hunk 17]
```diff
@@ -0,0 +1,251 @@
...
+library(lubridate)
+```
+
+# Connecting to RIMReP collection
+
+As mentioned above, we will connect to the AIMS Sea Surface Temperature Monitoring Program. This can take a minute or so.
+
+```{r connect}
+#Establishing connection
+data_bucket <- s3_bucket("s3://rimrep-data-public/091-aims-sst/test-50-64-spatialpart")
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_at_Site.Rmd]: Is this the correct bucket? Check with Leo

File: poc-data-api/Extracting_Water_Temperature_at_Site.Rmd [hunk 18]
```diff
@@ -0,0 +1,251 @@
...
+  
+**Note** that you can get additional information about the AIMS dataset by exploring its metadata as shown below.
+
+```{r metadata}
+#Checking metadata
+data_df$metadata
+```
+# Extracting variables of our interest from dataset
+
+We can extract data from the AIMS dataset by using `dplyr` verbs as shown below.
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_at_Site.Rmd]: We can extract location (coordinates) data from the AIMS dataset by using `dplyr` verbs as shown below.

File: poc-data-api/Extracting_Water_Temperature_at_Site.Rmd [hunk 19]
```diff
@@ -0,0 +1,251 @@
...
+
+We will use the `subsite` column to create the categories in our new column, as explained in the [exploring the dataset structure](#exploring-dataset-structure). If no condition is met, then we will label the row as `NA`.
+
+```{r dep_loc}
+sites <- sites %>% 
+  #Adding new column - Given categories based on a condition
+  mutate(deployment_location =  case_when(str_detect(subsite, "FL[0-9]{1}") ~ "reef flat",
+                                          str_detect(subsite, "SL[0-9]{1}") ~ "reef slope",
+                                          #If no condition is met, return NA
+                                          T ~ NA))
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_at_Site.Rmd]: `Other` instead of `NA`

File: poc-data-api/Extracting_Water_Temperature_at_Site.Rmd [hunk 20]
```diff
@@ -0,0 +1,251 @@
...
+
+#Checking results
+glimpse(sites)
+```
+## Extracting latitude and longitude values (Option #1)
+As explained [above](#exploring-dataset-structure), the `geometry` field is in WKB format, which we will transform into degree coordinates in the next step. Here, we will use the `WKB` to transform the `geometry` and then we will convert it to an `sf` object for easy data manipulation.
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_at_Site.Rmd]: we will use the `readWKB` function from `WKB` package to transform the `geometry`

File: poc-data-api/Extracting_Water_Temperature_at_Site.Rmd [hunk 21]
```diff
@@ -0,0 +1,251 @@
...
+```{r libraries, results = "hide", message = F, warning = F}
+library(arrow)
+library(dplyr)
+library(magrittr)
+library(stringr)
+library(wkb)
+library(ggplot2)
+library(sf)
+library(lubridate)
+```
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_at_Site.Rmd]: `library(rnaturalearth)`

File: poc-data-api/Extracting_Water_Temperature_at_Site.Rmd [hunk 22]
```diff
@@ -0,0 +1,251 @@
...
+  #Extracting latitude and longitude coordinates
+  select(site, lon, lat, deployment_location)
+
+#Checking results
+sites_coords
+```
+
+We can see that there are multiple coordinates for each site, but they are located in close proximity to one another. In this case, both sites include instrument deployments in the reef flat and slope. We will use this information to extract data from the original AIMS dataset.  
+  
+Note that `qc_val` is the variable containing the temperature data in this dataset.
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_at_Site.Rmd]: Note that `qc_val` is the variable containing quality-controlled temperature data in this dataset.

File: poc-data-api/Extracting_Water_Temperature_at_Site.Rmd [hunk 23]
```diff
@@ -0,0 +1,251 @@
...
+
+```{r extract}
+sites_temp <- data_df %>%
```
Comment (diodon) [inline poc-data-api/Extracting_Water_Temperature_at_Site.Rmd]: This next step takes a lot of time to execute. Would it be better if you extract the sites of interest based on the previous step and then do the calculation locally?

File: poc-data-api/useful_spatial_functions_data_extraction.R [hunk 24]
```diff
@@ -0,0 +1,145 @@
+# Useful functions to extract data using polygons
```
Comment (diodon) [inline poc-data-api/useful_spatial_functions_data_extraction.R]: maybe documenting each function using `roxygen2`. But this is not a priority right now.

**Review by diodon — CHANGES_REQUESTED** [review]

File: poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd (hunk 2 above)
Comment (lidefi87) [reply poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd]: Rephrased

File: poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd (hunk 5 above)
Comment (lidefi87) [reply poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd]: Added to list of imported packages

File: poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd (hunk 7 above)
Comment (lidefi87) [reply poc-data-api/Extracting_Spatial_Data_GBR_Features.Rmd]: added a new section (_workflow_) with these steps

File: poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd (hunk 8 above)
Comment (lidefi87) [reply poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd]: information added

File: poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd (hunk 9 above)
Comment (lidefi87) [reply poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd]: updated

File: poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd (hunk 13 above)
Comment (lidefi87) [reply poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd]: I added a `leaflet` map to the notebook. This way users can see how to produce an interactive and static map. We can remove the static version using `ggplot2` if you think is best

File: poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd (hunk 14 above)
Comment (lidefi87) [reply poc-data-api/Extracting_Water_Temperature_GBR_Features.Rmd]: I extracted the sites and then perform the calculations and it still takes some time, but it is less than the original version.

File: poc-data-api/Extracting_Water_Temperature_at_Site.Rmd (hunk 16 above)
Comment (lidefi87) [reply poc-data-api/Extracting_Water_Temperature_at_Site.Rmd]: This is my bad, I got this information from the [website](https://apps.aims.gov.au/metadata/view/4a12a8c0-c573-11dc-b99b-00008a07204e) describing the dataset and I did not actually check. A quick subset of the data using the maximum spatial extent gives me over 400 points. I have removed the number and rephrased this sentence.

File: poc-data-api/Extracting_Water_Temperature_at_Site.Rmd (hunk 17 above)
Comment (lidefi87) [reply poc-data-api/Extracting_Water_Temperature_at_Site.Rmd]: I checked in STAC and it shows the same URL.

File: poc-data-api/Extracting_Water_Temperature_at_Site.Rmd (hunk 23 above)
Comment (lidefi87) [reply poc-data-api/Extracting_Water_Temperature_at_Site.Rmd]: It still takes some time, but I think it is slightly faster now

File: poc-data-api/useful_spatial_functions_data_extraction.R (hunk 24 above)
Comment (lidefi87) [reply poc-data-api/useful_spatial_functions_data_extraction.R]: I have not used this library before, but I have been wanting to learn about it for a little while. Issue #8 opened. To be completed after notebooks are done.

**Comment by lidefi87** [comment]
@diodon final edits are now included in this pull request. Please let me know if you need any more changes.

**Review by diodon — APPROVED** [review]
